azurerm/data_source_batch_account.go

```diff
@@ -0,0 +1,71 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmBatchAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBatchAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
+			"location":            locationForDataSourceSchema(),
+			"storage_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"pool_allocation_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
since this is the Data Source this field isn't set-able, so we can remove the `Optional` here